bool contact
bool flight
bool impact
bool takeoff
